presto-atop/src/main/java/com/facebook/presto/atop/AtopMetadata.java

```diff
@@ -0,0 +1,184 @@
+        catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session,
+                                                            ConnectorTableHandle table,
+                                                            Constraint<ColumnHandle> constraint,
+                                                            Optional<Set<ColumnHandle>> desiredColumns)
+    {
+        AtopTableHandle tableHandle = checkType(table, AtopTableHandle.class, "table");
+        Optional<Map<ColumnHandle, Domain>> domains = constraint.getSummary().getDomains();
+        Domain endTimeDomain = Domain.all(TIMESTAMP);
+        Domain startTimeDomain = Domain.all(TIMESTAMP);
+        if (domains.isPresent()) {
+            if (domains.get().containsKey(START_TIME_HANDLE)) {
+                startTimeDomain = domains.get().get(START_TIME_HANDLE);
+            }
+            if (domains.get().containsKey(END_TIME_HANDLE)) {
+                endTimeDomain = domains.get().get(END_TIME_HANDLE);
+            }
+        }
+        AtopTableLayoutHandle layoutHandle = new AtopTableLayoutHandle(tableHandle, startTimeDomain, endTimeDomain);
+        ConnectorTableLayout tableLayout = getTableLayout(session, layoutHandle);
+        return ImmutableList.of(new ConnectorTableLayoutResult(tableLayout, constraint.getSummary()));
+    }
+
+    @Override
+    public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTableLayoutHandle tableLayoutHandle)
+    {
+        return new ConnectorTableLayout(tableLayoutHandle);
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        AtopTableHandle atopTableHandle = checkType(tableHandle, AtopTableHandle.class, "tableHandle");
+        ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
+        for (AtopColumn column : atopTableHandle.getTable().getColumns()) {
+            columns.add(new ColumnMetadata(column.getName(), typeManager.getType(column.getType())));
+        }
+        SchemaTableName schemaTableName = new SchemaTableName(atopTableHandle.getSchema(), atopTableHandle.getTable().getName());
+        return new ConnectorTableMetadata(schemaTableName, columns.build());
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, String schemaNameOrNull)
+    {
+        if (schemaNameOrNull == null) {
+            return Stream.of(AtopTable.values())
+                    .map(table -> new SchemaTableName(environment, table.getName()))
+                    .collect(Collectors.toList());
+        }
+        if (!listSchemaNames(session).contains(schemaNameOrNull)) {
+            return ImmutableList.of();
+        }
+        return Stream.of(AtopTable.values())
+                .map(table -> new SchemaTableName(schemaNameOrNull, table.getName()))
```

Comment:
It doesn't look like the Hive connector lower cases them, and mine are lower case anyway (it's just the label for passing to atop that's upper case)